appservice/message_send_checkpoint.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 Sumner Evans
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package appservice
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"maunium.net/go/mautrix/event"
+	"maunium.net/go/mautrix/id"
+)
+
+type MessageSendCheckpointStep string
+
+const (
+	ClientStep     MessageSendCheckpointStep = "CLIENT"
+	HomeserverStep MessageSendCheckpointStep = "HOMESERVER"
+	BridgeStep     MessageSendCheckpointStep = "BRIDGE"
+	RemoteStep     MessageSendCheckpointStep = "REMOTE"
+	CommandStep    MessageSendCheckpointStep = "COMMAND"
+)
+
+type MessageSendCheckpointStatus string
+
+const (
+	SuccesssStatus    MessageSendCheckpointStatus = "SUCCESS"
+	WillRetryStatus   MessageSendCheckpointStatus = "WILL_RETRY"
+	PermFailureStatus MessageSendCheckpointStatus = "PERM_FAILURE"
```

Comment:
`Step` and `Status` should be before the name of the step/status in the constant names